app/app.go

```diff
@@ -163,17 +164,27 @@ func (a *App) metricsReporterInterceptor(
 	case *pb.Event:
 		events = append(events, req.(*pb.Event))
 	case *pb.SendEventsRequest:
-		events = append(events, req.(*pb.SendEventsRequest).Events...)
-		retry = fmt.Sprintf("%d", req.(*pb.SendEventsRequest).Retry)
+		request := req.(*pb.SendEventsRequest)
+		events = append(events, request.Events...)
+		retry = fmt.Sprintf("%d", request.Retry)
 	default:
 		a.log.WithField("route", info.FullMethod).Infof("Unexpected request type %T", t)
 	}
-	
+
 	topic := events[0].Topic
 	l := a.log.
 		WithField("route", info.FullMethod).
 		WithField("topic", topic)
 
+	// using JSON to calculate the payload size introduces a small, invariant/constant error related when compared to
+	// the gRPC payload, but should be enough for observability
+	marshalledEvents, _ := json.Marshal(events)
```

Comment:
`protobuf` compacts field names. Serializing it into JSON will add field names. The JSON size will be greater than the wire size.

Reply:
on second tought, calculate the payload inside the type switch.
![image](https://github.com/topfreegames/eventsgateway/assets/14203456/581c449c-e85f-4b4d-adfc-965d69bc96b1)

Reply:
Nice catch! I've updated it

Reply:
👍